packages/cli/e2e-test/cli-e2e-test.js

```diff
@@ -250,5 +264,85 @@ async function testAppServe(pluginName, appDir) {
   }
 }
 
+/** Creates PG databases (drops if exists before) */
+async function createDB(database) {
+  const config = {
+    host: process.env.POSTGRES_HOST,
+    port: process.env.POSTGRES_PORT,
+    user: process.env.POSTGRES_USER,
+    password: process.env.POSTGRES_PASSWORD,
+  };
+
+  try {
+    await pgtools.dropdb({ config }, database);
+  } catch (_) {
+    /* do nothing*/
+  }
+  return pgtools.createdb(config, database);
+}
+
+/**
+ * Start serving the newly created backend and make sure that all db migrations works correctly
+ */
+async function testBackendStart(appDir, isPostgres) {
+  if (isPostgres) {
+    print('Creating DBs');
+    await Promise.all(
+      [
+        'catalog',
+        'scaffolder',
+        'auth',
+        'identity',
+        'proxy',
+        'techdocs',
+      ].map(name => createDB(`backstage_plugin_${name}`)),
+    );
+    print('Created DBs');
+  }
+
+  const child = spawnPiped(['yarn', 'workspace', 'backend', 'start'], {
+    cwd: appDir,
+  });
+
+  let stdout = '';
+  let stderr = '';
+  child.stdout.on('data', data => {
+    stdout = stdout + data.toString('utf8');
+  });
+  child.stderr.on('data', data => {
+    stderr = stderr + data.toString('utf8');
+  });
+  let successful = false;
+
+  try {
+    await waitFor(() => stdout.includes('Listening on ') || stderr !== '');
+    if (stderr !== '') {
+      // Skipping the whole block
+      throw new Error();
```

Comment:
Put something here ._.

Reply:
We are not catching this error right here, we just want to short-circuit the try-finally block. But the error message is in stderr, which I should use below

Reply:
It'll get caught and logged by `handleError`, so better to just put something to know that this is where we exited. Just something like `"Error received while running backend"`.